packages/vm/src/wasm_backend/engine.rs

```diff
@@ -54,6 +54,7 @@ pub fn make_compiling_engine(memory_limit: Option<Size>) -> Engine {
     #[cfg(not(feature = "cranelift"))]
     let mut compiler = Singlepass::default();
 
+    compiler.canonicalize_nans(true);
```

Comment:
Did you check if the tests fail when you disable this?

Reply:
Good idea! Yes, it produces a different hash